src/theory/arith/arith_rewriter.cpp

```diff
@@ -56,6 +57,62 @@ bool evaluateRelation(Kind rel, const L& l, const R& r)
   }
 }
 
+/** Flatten the given node (with child nodes of the same kind) into a vector */
+void flatten(TNode t, std::vector<TNode>& children)
+{
+  Kind k = t.getKind();
+  for (const auto& child : t)
+  {
+    if (child.getKind() == k)
+    {
+      flatten(child, children);
```

Comment:
Let's implement this in an iterative fashion instead of recursively. Also, I think that it would be good to just implement this in `node_algorithm.h` as a generic method.